modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/SnapshotRoutes.scala

```diff
@@ -40,78 +42,119 @@ final case class SnapshotRoutes[F[_]: Async: Hasher: KryoSerializer, S <: Snapsh
   implicit def binaryAndJsonEncoders[A <: AnyRef: Encoder]: List[EntityEncoder[F, A]] =
     List(BinaryCodec.encoder[F, A], CirceEntityEncoder.circeEntityEncoder[F, A])
 
+  private val serviceUnavailableNodeNotReady: F[Response[F]] =
+    ServiceUnavailable(("message" ->> s"Node should be in Ready state before fetching") :: HNil)
+
+  private val nodeStateF: F[NodeState] = nodeStorage.getNodeState
+  private def validStateForSnapshotReturn(state: NodeState): Boolean = state === NodeState.Ready
+
   protected val httpRoutes: HttpRoutes[F] = HttpRoutes.of[F] {
     case GET -> Root / "latest" / "ordinal" =>
-      import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
```

Comment:
are you sure you can remove these encoders?

Reply:
edit: probably yes 🤔 please double check if the endpoints return the binary format when required (I see there is a resolveEncoder part in some endpoints)

Reply:
I haven't removed it, I'm importing it globally now

Reply:
Yes, I know, but not every endpoint imported it. Having it imported globally makes it available for the other endpoints too, hence - worth checking whether it does not change the output format of those.

Reply:
Yeah I was thinking about it as well. We had a convention that we specify an encoder for every endpoint separately. But I guess we didn't have resolver for encoders back then. So I guess it should be fine, but would be good to check if we are not changing some logic (I guess if there are no clashes of implicits then it's ok)

Reply:
actually:
> (I guess if there are no clashes of implicits then it's ok)
it may be that we have to encoders (one imported globally, one resolved) -> @IPadawans check please if the code compiles 🤔

Reply:
It's compiling and it's working, I've tested
<img width="2056" alt="Captura de Tela 2024-02-19 às 10 26 33" src="https://github.com/Constellation-Labs/tessellation/assets/22091534/86154347-f81c-44d3-814e-62c184d7ab95">

Reply:
cool. please make sure the endpoints that are resolving the encoder (so pass `Accept: application/json` and `Accept: application/octet-stream` gets json/binary format. If that works, it's fine.

Reply:
<img width="2105" alt="Captura de Tela 2024-02-19 às 10 29 19" src="https://github.com/Constellation-Labs/tessellation/assets/22091534/81448c5b-ddfb-46a7-8ac3-3584cbc0f889">

<img width="2094" alt="Captura de Tela 2024-02-19 às 10 29 35" src="https://github.com/Constellation-Labs/tessellation/assets/22091534/35865c2f-7696-44a1-8a62-d30fbaad177b">

Reply:
/\ this is when the node is Ready

Reply:
cool. so it works :+1: